### DISCUSSION TITLE
Add Fantasy Strategy returning Exact GP

### DISCUSSION THREAD

**Discussion by GirishMKulkarni** [cornellius-gp/gpytorch#2040] [General]
I want to add new data sequentially to an already trained approximate GP model. I used the `get_fantasy_model` method in ApproximateGP to add the first new data point,  I get the Exact GP model in return. I want to know your thoughts on changing the return type. How can get the Approximate model even after adding new data? I need this because I want to have online training with one data point at a time.

**Comment by gpleiss** [comment]
It is probably possible to convert an exact GP model back to an approximate GP, but this is not implemented at the moment. This would probably require a research effort, and not something that we will be looking to implement in the near future.

**Comment by GirishMKulkarni** [comment]
Dear Geoff Pleiss,
I contacted you regarding an online version of the get_fantasy_model function for approximate GP. You mentioned that it will require some research efforts. I have been thinking about it and I think I might have a solution for it. Please see a very short description of the solution attached. Kindly let me know if you think this is a right way and can be pursued further. I will be happy to discuss/explore the details with you.

Best regards,
Girish

Dr. Girish Kulkarni
Postdoctoral Researcher
Mathematisches Institut der Heinrich-Heine-Unviersität Düsseldorf
Universitätsstr. 1
Building: 25.22.
Floor/Room: 03.51
D - 40225 Düsseldorf
